files/default/slurm/templates/slurm_parallelcluster_queue_gres.conf

```diff
@@ -5,10 +5,10 @@
         {% set dynamic_size = compute_resource_config.max_count - compute_resource_config.min_count %}
         {% set sanitized_instance_type = compute_resource_config.instance_type | sanify_instance_type %}
         {% if static_size > 0 %}
-NodeName={{ queue_name }}-st-{{ sanitized_instance_type }}-[1-{{ static_size }}] Name=gpu Type=tesla File=/dev/nvidia[0-{{ compute_resource_config.gpus - 1 }}]
+NodeName={{ queue_name }}-st-{{ sanitized_instance_type }}-[1-{{ static_size }}] Name=gpu Type={{ compute_resource_config.gpu_type }} File=/dev/nvidia[0-{{ compute_resource_config.gpus - 1 }}]
```

Comment:
moving from `tesla` to something like A100 can be a breaking change for users submitting with tesla as job requirement. We might be ok with that but let's review this choice once more.

Reply:
Also if we are using the GPU name from the describe-instance-types API be aware that some of them might have spaces in it (e.g. g4ad.4xlarge)
```
                    {
                        "Name": "Radeon Pro V520",
                        "Manufacturer": "AMD",
                        "Count": 1,
                        "MemoryInfo": {
                            "SizeInMiB": 8192
                        }
                    }
```

Shall we do a lowercase conversion + removal of whitespaces?